manubot/process/metadata.py

```diff
@@ -157,3 +157,53 @@ def get_manuscript_urls(html_url: Optional[str] = None) -> dict:
             + "\n".join(x.url for x in response.history + [response])
         )
     return urls
+
+
+def get_software_versions():
+    from manubot import __version__ as manubot_version
+
+    return {
+        "manubot_version": manubot_version,
+        "rootstock_commit": get_rootstock_commit(),
+    }
+
+
+def get_rootstock_commit():
+    """
+    Return the most recent commit in common between the git repository
+    this function is run within (usually a Manubot manuscript repository)
+    and the `master` branch of the `rootstock` remote.
+
+    WARNING: This function may modify the git repository its executed within:
```

Comment:
How conservative do we want to be regarding modifying the local git repository?  This approach is reasonable.  An even more conservative approach would be to only run this if 1) the `rootstock` remote already exists or 2) Manubot is being run in a CI service.

Reply:
> How conservative do we want to be regarding modifying the local git repository? 

I don't think it's too damaging to create the `rootstock` remote if it doesn't exist. I think it's important that we keep the functionality for local builds not just CI. I will think if there's a better way we can ensure `rootstock` remote is only added if the repo seems to be a manubot manuscript.

Reply:
Sure, that makes sense.  I wanted to open the discussion but don't have any objections.

This being part of the `process` subcommand may already be sufficient for making sure it is only run as part of a Manubot manuscript.

Reply:
I think we can proceed with the current implementation and continue to think if there is a more targeted solution.